src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java

```diff
@@ -104,9 +109,35 @@ public void construct() {
     hotNodeClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
     hotNodeClusterRca.addAllUpstreams(Collections.singletonList(hotJVMNodeRca));
 
+    constructShardResourceUsageGraph();
+
     // constructResourceHeatMapGraph();
   }
 
+  private void constructShardResourceUsageGraph() {
+    Metric cpuUsage = new CPU_Utilization(5);
+    Metric ioTotThroughput = new IO_TotThroughput(5);
+    Metric ioTotSyscallRate = new IO_TotalSyscallRate(5);
+
+    cpuUsage.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    ioTotThroughput.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    ioTotSyscallRate.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(cpuUsage);
+    addLeaf(ioTotThroughput);
+    addLeaf(ioTotSyscallRate);
+
+    // High CPU Usage RCA
+    HighCPUShardRca highCPUShardRca = new HighCPUShardRca(5, 12, cpuUsage, ioTotThroughput, ioTotSyscallRate);
+    highCPUShardRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
```

Comment:
If we specify `LOCUS_DATA_MASTER_NODE` for a graph node then the graph node gets executed on a data node and master node, right ? Here we have this in case the cluster does not have dedicated masters ?

Reply:
The downside is we will waste cycles executing this graph node on a dedicated master node, where there will be no shards. I guess we don't have an appropriate tags to distinguish between the two. We should as a backlog issues.

Reply:
> If we specify `LOCUS_DATA_MASTER_NODE` for a graph node then the graph node gets executed on a data node and master node, right ? Here we have this in case the cluster does not have dedicated masters ?

Correct. This is meant for the case where the clusters isn't configured with dedicated masters.

Reply:
> The downside is we will waste cycles executing this graph node on a dedicated master node, where there will be no shards. I guess we don't have an appropriate tags to distinguish between the two. We should as a backlog issues.

Correct, we will create a issue for this.